File: java/09-reflection/Level9.java
public class Level9 {
    public static void main(String[] args) throws Exception {
        String className = "java.lang.StringBuilder";
        Class<?> cls = Class.forName(className);
        Object obj = cls.getDeclaredConstructor().newInstance();

        cls.getMethod("append", String.class).invoke(obj, "Reflection says hi!");
        String result = (String) cls.getMethod("toString").invoke(obj);
        System.out.println(result);
    }
}

File: c-sharp/09-reflection/Level9.cs
using System;
using System.Reflection;

string className = "System.Text.StringBuilder";
Type? type = Type.GetType(className);

if (type != null)
{
    object? instance = Activator.CreateInstance(type);
    MethodInfo? appendMethod = type.GetMethod("Append", new[] { typeof(string) });
    appendMethod?.Invoke(instance, new object[] { "Reflection says hi!" });

    MethodInfo? toStringMethod = type.GetMethod("ToString", Type.EmptyTypes);
    string? result = toStringMethod?.Invoke(instance, null) as string;
    Console.WriteLine(result);
}
